modin/experimental/core/storage_formats/hdk/query_compiler.py

```diff
@@ -657,11 +665,20 @@ def mul(self, other, **kwargs):
         return self._bin_op(other, "mul", **kwargs)
 
     def mod(self, other, **kwargs):
-        int_codes = np.typecodes["AllInteger"]
-        if all(t.char in int_codes for t in self._modin_frame.dtypes):
-            return self._bin_op(other, "mod", **kwargs)
-        else:
-            raise NotImplementedError("Non-integer operands in modulo operation")
+        def check_int(obj):
+            if isinstance(obj, DFAlgQueryCompiler):
+                int_codes = np.typecodes["AllInteger"]
+                cond = all(t.char in int_codes for t in obj._modin_frame.dtypes)
```

Comment:
may we better use [`is_integer_dtype()`](https://pandas.pydata.org/docs/reference/api/pandas.api.types.is_integer_dtype.html) here?